### DISCUSSION TITLE
Error with package: pip install git+https://github.com/AI4Finance-Foundation/FinRL.git

### DISCUSSION THREAD

**Discussion by adamcoz** [AI4Finance-Foundation/FinRL#1105] [General]
I am getting the error message whilst attempting to follow through the 'FinRL_stock_trading_fundamental' notebook in colab. Can anyone advise on a solution? 

Error Message:
"
Collecting git+https://github.com/AI4Finance-LLC/FinRL-Library.git
  Cloning https://github.com/AI4Finance-LLC/FinRL-Library.git to /tmp/pip-req-build-9ce30_jx
  Running command git clone --filter=blob:none --quiet https://github.com/AI4Finance-LLC/FinRL-Library.git /tmp/pip-req-build-9ce30_jx
  Resolved https://github.com/AI4Finance-LLC/FinRL-Library.git to commit e85aaab4ede4d8ed0a245da967c10b6a1cd66d7d
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting elegantrl@ git+https://github.com/AI4Finance-Foundation/ElegantRL.git#egg=elegantrl (from finrl==0.3.6)
  Cloning https://github.com/AI4Finance-Foundation/ElegantRL.git to /tmp/pip-install-v_9y0fh0/elegantrl_50d32d2d39c6404880e5e902ad175d52
  Running command git clone --filter=blob:none --quiet https://github.com/AI4Finance-Foundation/ElegantRL.git /tmp/pip-install-v_9y0fh0/elegantrl_50d32d2d39c6404880e5e902ad175d52
  Resolved https://github.com/AI4Finance-Foundation/ElegantRL.git to commit 4c1893aeeb350e952bdcce916b13e6650a84cd68
  Preparing metadata (setup.py) ... done
Collecting alpaca-trade-api<4,>=3 (from finrl==0.3.6)
  Using cached alpaca_trade_api-3.0.2-py3-none-any.whl (34 kB)
Collecting ccxt<4,>=3 (from finrl==0.3.6)
  Using cached ccxt-3.1.60-py2.py3-none-any.whl (4.0 MB)
Collecting exchange-calendars<5,>=4 (from finrl==0.3.6)
  Using cached exchange_calendars-4.5-py3-none-any.whl (191 kB)
Collecting jqdatasdk<2,>=1 (from finrl==0.3.6)
  Using cached jqdatasdk-1.9.0-py3-none-any.whl (165 kB)
Collecting pyfolio<0.10,>=0.9 (from finrl==0.3.6)
  Using cached pyfolio-0.9.2.tar.gz (91 kB)
  Preparing metadata (setup.py) ... done
Collecting pyportfolioopt<2,>=1 (from finrl==0.3.6)
  Using cached pyportfolioopt-1.5.5-py3-none-any.whl (61 kB)
Collecting ray[default,tune]<3,>=2 (from finrl==0.3.6)
  Using cached ray-2.7.0-cp310-cp310-manylinux2014_x86_64.whl (62.5 MB)
Requirement already satisfied: scikit-learn<2,>=1 in /usr/local/lib/python3.10/dist-packages (from finrl==0.3.6) (1.2.2)
Collecting stable-baselines3[extra]>=2.0.0a5 (from finrl==0.3.6)
  Using cached stable_baselines3-2.1.0-py3-none-any.whl (178 kB)
Collecting stockstats<0.6,>=0.5 (from finrl==0.3.6)
  Using cached stockstats-0.5.4-py2.py3-none-any.whl (21 kB)
Collecting wrds<4,>=3 (from finrl==0.3.6)
  Using cached wrds-3.1.6-py3-none-any.whl (12 kB)
Requirement already satisfied: yfinance<0.3,>=0.2 in /usr/local/lib/python3.10/dist-packages (from finrl==0.3.6) (0.2.28)
Requirement already satisfied: pandas>=0.18.1 in /usr/local/lib/python3.10/dist-packages (from alpaca-trade-api<4,>=3->finrl==0.3.6) (1.5.3)
Requirement already satisfied: numpy>=1.11.1 in /usr/local/lib/python3.10/dist-packages (from alpaca-trade-api<4,>=3->finrl==0.3.6) (1.23.5)
Requirement already satisfied: requests<3,>2 in /usr/local/lib/python3.10/dist-packages (from alpaca-trade-api<4,>=3->finrl==0.3.6) (2.31.0)
Collecting urllib3<2,>1.24 (from alpaca-trade-api<4,>=3->finrl==0.3.6)
  Using cached urllib3-1.26.16-py2.py3-none-any.whl (143 kB)
Requirement already satisfied: websocket-client<2,>=0.56.0 in /usr/local/lib/python3.10/dist-packages (from alpaca-trade-api<4,>=3->finrl==0.3.6) (1.6.2)
Collecting websockets<11,>=9.0 (from alpaca-trade-api<4,>=3->finrl==0.3.6)
  Using cached websockets-10.4-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (106 kB)
Collecting msgpack==1.0.3 (from alpaca-trade-api<4,>=3->finrl==0.3.6)
  Using cached msgpack-1.0.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (323 kB)
Collecting aiohttp==3.8.2 (from alpaca-trade-api<4,>=3->finrl==0.3.6)
  Using cached aiohttp-3.8.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.0 MB)
Collecting PyYAML==6.0 (from alpaca-trade-api<4,>=3->finrl==0.3.6)
  Using cached PyYAML-6.0-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (682 kB)
Collecting deprecation==2.1.0 (from alpaca-trade-api<4,>=3->finrl==0.3.6)
  Using cached deprecation-2.1.0-py2.py3-none-any.whl (11 kB)
Requirement already satisfied: attrs>=17.3.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp==3.8.2->alpaca-trade-api<4,>=3->finrl==0.3.6) (23.1.0)
Collecting charset-normalizer<3.0,>=2.0 (from aiohttp==3.8.2->alpaca-trade-api<4,>=3->finrl==0.3.6)
  Using cached charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Collecting multidict<6.0,>=4.5 (from aiohttp==3.8.2->alpaca-trade-api<4,>=3->finrl==0.3.6)
  Using cached multidict-5.2.0-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (175 kB)
Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in /usr/local/lib/python3.10/dist-packages (from aiohttp==3.8.2->alpaca-trade-api<4,>=3->finrl==0.3.6) (4.0.3)
Requirement already satisfied: yarl<2.0,>=1.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp==3.8.2->alpaca-trade-api<4,>=3->finrl==0.3.6) (1.9.2)
Requirement already satisfied: frozenlist>=1.1.1 in /usr/local/lib/python3.10/dist-packages (from aiohttp==3.8.2->alpaca-trade-api<4,>=3->finrl==0.3.6) (1.4.0)
Requirement already satisfied: aiosignal>=1.1.2 in /usr/local/lib/python3.10/dist-packages (from aiohttp==3.8.2->alpaca-trade-api<4,>=3->finrl==0.3.6) (1.3.1)
Requirement already satisfied: packaging in /usr/local/lib/python3.10/dist-packages (from deprecation==2.1.0->alpaca-trade-api<4,>=3->finrl==0.3.6) (23.1)
Requirement already satisfied: setuptools>=60.9.0 in /usr/local/lib/python3.10/dist-packages (from ccxt<4,>=3->finrl==0.3.6) (67.7.2)
Requirement already satisfied: certifi>=2018.1.18 in /usr/local/lib/python3.10/dist-packages (from ccxt<4,>=3->finrl==0.3.6) (2023.7.22)
Requirement already satisfied: cryptography>=2.6.1 in /usr/local/lib/python3.10/dist-packages (from ccxt<4,>=3->finrl==0.3.6) (41.0.3)
Collecting aiodns>=1.1.1 (from ccxt<4,>=3->finrl==0.3.6)
  Using cached aiodns-3.0.0-py3-none-any.whl (5.0 kB)
Collecting pyluach (from exchange-calendars<5,>=4->finrl==0.3.6)
  Using cached pyluach-2.2.0-py3-none-any.whl (25 kB)
Requirement already satisfied: python-dateutil in /usr/local/lib/python3.10/dist-packages (from exchange-calendars<5,>=4->finrl==0.3.6) (2.8.2)
Requirement already satisfied: toolz in /usr/local/lib/python3.10/dist-packages (from exchange-calendars<5,>=4->finrl==0.3.6) (0.12.0)
Collecting tzdata (from exchange-calendars<5,>=4->finrl==0.3.6)
  Using cached tzdata-2023.3-py2.py3-none-any.whl (341 kB)
Collecting korean-lunar-calendar (from exchange-calendars<5,>=4->finrl==0.3.6)
  Using cached korean_lunar_calendar-0.3.1-py3-none-any.whl (9.0 kB)
Requirement already satisfied: six in /usr/local/lib/python3.10/dist-packages (from jqdatasdk<2,>=1->finrl==0.3.6) (1.16.0)
Requirement already satisfied: SQLAlchemy>=1.2.8 in /usr/local/lib/python3.10/dist-packages (from jqdatasdk<2,>=1->finrl==0.3.6) (2.0.20)
Collecting thriftpy2>=0.3.9 (from jqdatasdk<2,>=1->finrl==0.3.6)
  Using cached thriftpy2-0.4.16.tar.gz (643 kB)
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  Preparing metadata (setup.py) ... error
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
"

Thanks in advance.

**Comment by mmmarchetti** [comment]
This error has already been addressed on the issues page. It is an error caused by the thriftpy2 library inside colab during installation. A viable alternative is to use it locally as it works correctly.
[Issue Page Error details.](https://github.com/AI4Finance-Foundation/FinRL/issues/1100)
